src/Generating/FinishGen.cpp

```diff
@@ -337,7 +340,7 @@ void cFinishGenTallGrass::GenFinish(cChunkDesc & a_ChunkDesc)
 			{
 				a_ChunkDesc.SetBlockTypeMeta(x, y, z, E_BLOCK_TALL_GRASS, 2);
 			}
-			else
+			else if ((biome != biIcePlains) && (biome != biColdBeach) && (biome != biIceMountains) && (biome != biIcePlainsSpikes))
```

Comment:
I believe we have a `IsBiomeVeryCold` function for this. It's defined in BiomeDef I believe.